### ISSUE TITLE
please add Linux build doc

### ISSUE THREAD

**Issue by j2l** [Nor-s/Anim#21]
**Is your feature request related to a problem? Please describe.**
cd build
cmake ..
```
Using CMake version 3.30.5
CMAKE_RUNTIME_OUTPUT_DIRECTORY : 
CMAKE_BINARY_DIR : /home/pm/Documents/github/Anim-holistic/build
==========================================================
Build python
CMake Error at CMakeLists.txt:70 (file):
  file COPY cannot find "/home/pm/Documents/github/Anim-holistic/python": No
  such file or directory.


CMake Error at CMakeLists.txt:71 (file):
  file COPY cannot find
  "/home/pm/Documents/github/Anim-holistic/python/python310.dll": No such
  file or directory.


CMake Error at CMakeLists.txt:72 (file):
  file COPY cannot find
  "/home/pm/Documents/github/Anim-holistic/python/python3.dll": No such file
  or directory.


==========================================================
Compiler
--  - ID       	: GNU
--  - Version  	: 11.4.0
--  - Path     	: /usr/bin/c++
GLAD
CMake Deprecation Warning at external/glad/CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


STB
CMake Deprecation Warning at external/imgui/CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


CMake Error at CMakeLists.txt:113 (add_subdirectory):
  The source directory

    /home/pm/Documents/github/Anim-holistic/external/assimp

  does not contain a CMakeLists.txt file.


CMake Error at CMakeLists.txt:114 (add_subdirectory):
  The source directory

    /home/pm/Documents/github/Anim-holistic/external/glfw

  does not contain a CMakeLists.txt file.


hiredis
CMake Error at CMakeLists.txt:118 (add_subdirectory):
  The source directory

    /home/pm/Documents/github/Anim-holistic/external/hiredis

  does not contain a CMakeLists.txt file.


jsoncpp
CMake Error at CMakeLists.txt:120 (add_subdirectory):
  The source directory

    /home/pm/Documents/github/Anim-holistic/external/jsoncpp

  does not contain a CMakeLists.txt file.


GLM
CMake Error at CMakeLists.txt:122 (add_subdirectory):
  The source directory

    /home/pm/Documents/github/Anim-holistic/external/glm

  does not contain a CMakeLists.txt file.


Python/home/pm/Documents/github/Anim-holistic/build/bin/python/
CMake Error at /home/pm/.local/lib/python3.10/site-packages/cmake/data/share/cmake-3.30/Modules/FindPackageHandleStandardArgs.cmake:233 (message):
  Could NOT find Python3 (missing: Python3_INCLUDE_DIRS Python3_LIBRARIES
  Development Development.Module Development.Embed) (found suitable exact
  version "3.10.12")
Call Stack (most recent call first):
  /home/pm/.local/lib/python3.10/site-packages/cmake/data/share/cmake-3.30/Modules/FindPackageHandleStandardArgs.cmake:603 (_FPHSA_FAILURE_MESSAGE)
  /home/pm/.local/lib/python3.10/site-packages/cmake/data/share/cmake-3.30/Modules/FindPython/Support.cmake:4001 (find_package_handle_standard_args)
  /home/pm/.local/lib/python3.10/site-packages/cmake/data/share/cmake-3.30/Modules/FindPython3.cmake:602 (include)
  CMakeLists.txt:130 (find_package)


-- Configuring incomplete, errors occurred!
```

**Describe the solution you'd like**
a working Anim 0.10.0 :smile: 

**Additional context**
using a virtualenv for Anim: vanim
ubuntu 22.04

**Comment by Nor-s** [comment]
sorry. When I changed to CPython, I stopped caring about Linux. I'll try setting up the environment again soon. thank you

**Comment by j2l** [comment]
@Nor-s thank you for your prompt reply and your willingness to support Linux.
Could you please update here when you update?
Thank you again

**Comment by Nor-s** [comment]
<img width="706" alt="image" src="https://github.com/user-attachments/assets/039b1017-ec8f-47c9-b08e-c22f2175d790">

@j2l 
Hello, the Linux build now works in my environment. 
Could you please check? Thank you.
And, thank you for your curiosity about my old project.

- Ubuntu 22.04.5 LTS (GNU/Linux 5.15.153.1-microsoft-standard-WSL2 x86_64)
- gcc 11.4.0 x86_64-linux-gnu
```
# maybe you need
sudo apt update
sudo apt install libssl-dev
sudo apt install zlib1g-dev
sudo apt install libffi-dev
sudo apt-get install redis-server

chmod +x ./scripts/build_python_linux.sh
```

```
git clone --recursive https://github.com/Nor-s/Anim.git
cd Anim
mkdir build
cd build
cmake ..
make
cd bin
./Anim
```

**Comment by j2l** [comment]
GREAT NEWS!
Thank you very much @Nor-s !!

For the records, at `pip install`:
```
ERROR: Could not find a version that satisfies the requirement pywin32==304 (from versions: none)

[notice] A new release of pip is available: 24.2 -> 24.3.1
[notice] To update, run: pip install --upgrade pip
ERROR: No matching distribution found for pywin32==304
...
ERROR: Cannot install -r ./scripts/requirements.txt (line 20) and protobuf==3.20.1 because these package versions have conflicting dependencies.

The conflict is caused by:
    The user requested protobuf==3.20.1
    mediapipe 0.10.14 depends on protobuf<5 and >=4.25.3

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip to attempt to solve the dependency conflict
```
`cmake ..` works
`make`:
```
97%] Building CXX object src/CMakeFiles/Anim.dir/UI/component_layer.cpp.o
/mnt/phil/Anim/src/UI/component_layer.cpp: In member function ‘void ui::ComponentLayer::draw(ui::ComponentContext&, Scene*)’:
/mnt/phil/Anim/src/UI/component_layer.cpp:31:44: warning: format not a string literal and no format arguments [-Wformat-security]
   31 |                                 ImGui::Text(entity->get_name().data());
      |                                 ~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~
[ 98%] Building CXX object src/CMakeFiles/Anim.dir/UI/scene_layer.cpp.o
[ 98%] Building CXX object src/CMakeFiles/Anim.dir/UI/text_edit_layer.cpp.o
[ 98%] Building CXX object src/CMakeFiles/Anim.dir/UI/timeline_layer.cpp.o
[ 98%] Building CXX object src/CMakeFiles/Anim.dir/UI/main_layer.cpp.o
[ 99%] Building CXX object src/CMakeFiles/Anim.dir/UI/imgui_helper.cpp.o
/mnt/phil/Anim/src/UI/imgui_helper.cpp: In function ‘bool ui::DragFPropertyXYZ(const char*, float*, float, float, float, const char*, const string&, int)’:
/mnt/phil/Anim/src/UI/imgui_helper.cpp:46:20: warning: format not a string literal and no format arguments [-Wformat-security]
   46 |         ImGui::Text(label);
      |         ~~~~~~~~~~~^~~~~~~
[ 99%] Building CXX object src/CMakeFiles/Anim.dir/event/event_history.cpp.o
[ 99%] Building CXX object src/CMakeFiles/Anim.dir/app.cpp.o
[ 99%] Building CXX object src/CMakeFiles/Anim.dir/scene/main_scene.cpp.o
[100%] Linking CXX executable ../bin/Anim
[100%] Built target Anim
```
but it WORKS!
